llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -5288,6 +5288,54 @@ Value *llvm::simplifyExtractElementInst(Value *Vec, Value *Idx,
   return ::simplifyExtractElementInst(Vec, Idx, Q, RecursionLimit);
 }
 
+static Value *getCommonPHIValue(Value *PreviousCommon, Value *Current,
+                                const SimplifyQuery &Q) {
+  if (!PreviousCommon)
+    return Current;
+
+  if (PreviousCommon == Current)
+    return PreviousCommon;
+
+  Constant *PreviousCommonC, *CurrentC;
+  if (match(PreviousCommon, m_Constant(PreviousCommonC)) &&
+      match(Current, m_Constant(CurrentC))) {
+    auto *VecTy = dyn_cast<FixedVectorType>(PreviousCommonC->getType());
+    if (VecTy) {
+      SmallVector<Constant *> NewCommonC;
+      unsigned NumElts = VecTy->getNumElements();
+      for (unsigned I = 0; I != NumElts; ++I) {
+        Constant *PrevElt = PreviousCommonC->getAggregateElement(I);
+        Constant *CurrElt = CurrentC->getAggregateElement(I);
+        if (!PrevElt || !CurrElt)
+          return nullptr;
```

Comment:
Can these actually fail at this point?

Reply:
If `m_Constant` matches a `ConstantExpr` I think.

Reply:
Add some tests with unresolvable expressions?

Reply:
I've added some tests with constant expressions that survive simplification. These do indeed result in `!PrevElt` or `!CurrEllt` here.